### ISSUE TITLE
💅 Vue Router's internal <Component> is considered as unused

### ISSUE THREAD

**Issue by hahavenn** [biomejs/biome#11230]
### Environment information

According to vue-router docs https://router.vuejs.org/guide/advanced/router-view-slot#RouterView-slot we can do:
```vue
<router-view v-slot="{ Component }">
  <component :is="Component" />
</router-view>
```

But biome considered `<Component>` as unused. Also we can't suppress this error with html comment

### Rule name

noUndeclaredVariables

### Playground link

https://biomejs.dev/playground/?lintRules=noUndeclaredVariables&tab=formatter&pane=Diagnostics&code=PAB0AGUAbQBwAGwAYQB0AGUAPgAKACAAIAA8ACEALQAtACAAYgBpAG8AbQBlAC0AaQBnAG4AbwByAGUAIABsAGkAbgB0AC8AYwBvAHIAcgBlAGMAdABuAGUAcwBzAC8AbgBvAFUAbgBkAGUAYwBsAGEAcgBlAGQAVgBhAHIAaQBhAGIAbABlAHMAIAAtAC0APgAKACAAIAA8AFIAbwB1AHQAZQByAFYAaQBlAHcAIAB2AC0AcwBsAG8AdAA9ACIAewAgAEMAbwBtAHAAbwBuAGUAbgB0ACAAfQAiAD4ACgAgACAAIAAgADwASwBlAGUAcABBAGwAaQB2AGUAIAA6AGkAbgBjAGwAdQBkAGUAPQAiACcARwByAGUAZQB0AGkAbgBnAHMAJwAiAD4ACgAgACAAIAAgACAAIAA8AGMAbwBtAHAAbwBuAGUAbgB0ACAAOgBpAHMAPQAiAEMAbwBtAHAAbwBuAGUAbgB0ACIAIAAvAD4ACgAgACAAIAAgADwALwBLAGUAZQBwAEEAbABpAHYAZQA%2BAAoAIAAgADwALwBSAG8AdQB0AGUAcgBWAGkAZQB3AD4ACgA8AC8AdABlAG0AcABsAGEAdABlAD4A&language=vue

### Expected result

It shouldn't throw error

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
